### PR TITLE
libtool: modernize

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libtool/2.4.6**

For https://github.com/conan-io/hooks/pull/320

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
duplicate of https://github.com/conan-io/conan-center-index/pull/6875 ;)

**Comment by madebr** [comment]
@SpaceIm 
Indeed.
Can you also include my change of datarootdir to the `res` folder?

File: recipes/libtool/all/conanfile.py [hunk 1]
```diff
@@ -224,7 +228,7 @@ def package_info(self):
             self.output.info("Setting {} environment variable to {}".format(key, value))
             setattr(self.env_info, key, value)
 
-        libtool_aclocal = tools.unix_path(os.path.join(self.package_folder, "bin", "share", "aclocal"))
+        libtool_aclocal = tools.unix_path(os.path.join(self.package_folder, "res", "aclocal"))
```
Comment (madebr) [inline recipes/libtool/all/conanfile.py:231]:
```suggestion
        libtool_aclocal = tools.unix_path(os.path.join(self._datarootdir, "aclocal"))
```

**Review by Croydon — APPROVED** [review]

**Comment by SpaceIm** [comment]
> @SpaceIm
> Indeed.
> Can you also include my change of datarootdir to the `res` folder?

Yep, done

**Pull request closed without merging**